File: 01/circle_area.rb
#! /usr/bin/ruby
# circle_area.rb computes the area of a circle given its radius
# 
# Input: the radius of a circle
# Precondition: the radius is not negative
# Output: the area of the circle.
#
# Begun by: Prof. Adams, for CS 214 at Calvin College.
# Completed by: Lily McAboy
# Date: 1/19/2024
###############################################################

# function circleArea returns a circle's area, given its radius
# Parameters: r, a number
# Precondition: r > 0.
# Returns: the area of a circle whose radius is r.
PI = 3.14159267 # set a constant pi  
def circleArea(r) #create a function called circleArea that takes in r (radius)
    PI * r ** 2 #Compute the area of a circle
end #end the function

if __FILE__ == $0 #_FILE_ holds the filename where we have our code, $0 is the name of the file to start the program
   puts "To compute the area of a circle," #print this statement
   print " enter its radius: " #Ask for an input 
   radius = gets.chomp.to_f #given a radius from the user, copy it to the program
   print "The circle's area is: " #print out 
   puts circleArea(radius) # print area of circle
end # end

File: 01/clojure/src/circle_area.clj

;;;; circle_area.clj calculates the area of a circle.
;;;;
;;;; Input: The radius of a circle.
;;;; Output: The area of that circle.
;;;;
;;;; Usage: clojure -m circle_area
;;;;
;;;; Begun by: Prof. Adams, for CS 214 at Calvin College.
;;;; Completed by: Lily McAboy
;;;; Date: /19/2024
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns circle_area) ; namespace function names the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function circleArea() computes the area of a circle.
;;; Receive: itsRadius, a number.
;;; Precondition: itsRadius >= 0.0.
;;; Return: the area of the corresponding circle.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn circleArea [itsRadius] ;;creating a function called circleArea and taking in itsradius to compute the circle's area
  (* Math/PI (* itsRadius itsRadius) )  ; return PI*r^2
)

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function -main is where execution begins
;;; Input: the radius of a circle.
;;; Output: the area of that circle.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
  (println "\nTo compute the area of a circle,") 
  (print   " enter its radius: ") (flush) ;;flush is the output stream
  (let
    [ radius (read) ] ;;read IN radius from user

    (assert (>= radius 0) "-main: radius must be positive")
    (printf "\nThe area is %f\n\n" (circleArea radius))

    (print "\nThe area is ")
    (print (circleArea radius))
    (print "\n\n")

    (printf "\nThe area is %.15f\n\n" (circleArea radius))
  )
)
